pkg/granted/registry/add.go

```diff
@@ -76,84 +77,163 @@ var AddCommand = cli.Command{
 			PrefixAllProfiles:       prefixAllProfiles,
 		}
 
-		registry, err := gitregistry.New(gitregistry.Opts{
-			Name:         name,
-			URL:          gitURL,
-			Path:         pathFlag,
-			Filename:     configFileName,
-			RequiredKeys: requiredKey,
-			Interactive:  true,
-		})
+		if IsGitRepository(URL) {
```

Comment:
The `IsGitRepository` check here is a bit brittle - it's possible that someone could be self-hosting a git repo at a URL which isn't GitLab or GitHub (or even simply using Azure DevOps, or Bitbucket etc).

Rather than trying to infer if it's a git repo or using our RPC protocol automatically, let's add a `--type` flag for users to indicate which type of registry they want to add.

e.g.
```
# add a HTTP registry using the granted.registry.aws.v1alpha1 protocol
granted registry add --type http --url https://commonfate.example.com

# add a git registry
granted registry add --type git --url https://some.git.repository.url
```

We should default to `--type=git` to preserve the current behaviour.